### PR TITLE
Add Uint512DivMod Hint

### PR DESCRIPTION
**Pull request by mmsc2**
# TITLE

## Description

Description of the pull request changes and motivation.

## Checklist
- [x] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [x] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

**Review by fmoletta — APPROVED** [review]
I would suggest adding an integration test or at least creating an issue for it, otherwise lgtm

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #### Upcoming Changes
 
+* feat: Implement 'Uint512DivMod' hint [#1151](https://github.com/lambdaclass/cairo-rs/pull/1171)
```
Comment (fmoletta) [inline CHANGELOG.md:5]:
```suggestion
* feat: Implement 'Uint512DivMod' hint [#1171](https://github.com/lambdaclass/cairo-rs/pull/1171)
```

**Comment by mmsc2** [comment]
> I would suggest adding an integration test or at least creating an issue for it, otherwise lgtm

yes! I am working on it

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 #### Upcoming Changes
 
+* feat: Implement 'Uint512DivMod' hint [#1171](https://github.com/lambdaclass/cairo-rs/pull/1171)
```
Comment (pefontana) [inline CHANGELOG.md:5]: just a detail, the implementation of the cairo 1 hints are mentioned above in the changelog.
Just add the link to this PR in that line and remove this line

**Review by pefontana — COMMENTED** [review]
Nice one!
As Fede mentioned, add an integration test and we are good to go

File: cairo_programs/cairo-1-contracts/uint512_div_mod.cairo [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        assert(r == 0, '1 % 1 != 0');
+
+        let two = u512 {limb0: 0, limb1: 0, limb2: 0, limb3: 2};
+        let (q, r) = u512_safe_div_rem_by_u256(two, integer::u256_as_non_zero(1));
+        assert(q == two, '2/1 != 2');
+        assert(r == 0, '2/1 != 0');
+
+        return ();
+    }
+}
```
Comment (fmoletta) [inline cairo_programs/cairo-1-contracts/uint512_div_mod.cairo:26]:
```suggestion
}

```
Add newline at EOF

**Review by pefontana — APPROVED** [review]

**Pull request merged**